.circleci/config.yml

```diff
@@ -23,6 +23,8 @@ jobs:
   build:
     environment:
       - TZ: "/usr/share/zoneinfo/America/Los_Angeles"
+      # Change this to be the repository with the notebook you want to build
+      - REPO_NAME: https://github.com/norvig/pytudes
```

Comment:
Shouldn't this be set to https://github.com/binder-examples/continuous-build? At least I was assuming that we'd use this repo (and its contents) as demo for having a repo that uses the CI setup. Instead of using this repo to build a different repo.

Reply:
We definitely could, in which case I would undo this PR because the repo build wouldn't fail.  The default used to be the repo it was building from but after the requested change I made the default the one set in the example.

Reply:
Given preference for not having the repo here, I set it to be that repo for two reasons:

 - to show that it doesn't have to be the same repo serving it
 - to match the example provided in the docs.

It would be good to discuss 1) if the default should assume the repo serving should also provide it (and thus fail if this doesn't happen, or default to the pytudes above) or 2) if the default should be to just assume the user won't connect anything, serve the pytudes, and then the user would obviously notice this, and change it (and have no fail). I can see arguments for both, and I think you guys should decide :)

Reply:
I would lean toward the CI building the image for this repo itself. I think the build repo that builds other images is a valid and useful case, but targeting these demo repos as self-contained makes sense to me.

Then we can make sure the README explains that this is an example for repos meant to work with repo2docker to get CI testing for themselves.

Reply:
Keeping the number of irrelevant files small is 👍. `repo2docker` will successfully build an empty repo so we don't have to add anything beyond the CI configurations. Maybe to make it a bit realistic I'd add a `requirements.txt` and a `somethingsomething-here.ipynb` that uses a library mentioned in the requirements. Then we can use the notebook as an example of "execute something in the repo as part of building it".

My feeling/assumption is that most people will want to use `repo2docker` as part of the CI of a repository to self-test that their repository will continue to work with BinderHub.

Because `repo2docker` will build an empty repo there aren't any required files we can look for to decide when to trigger the "build a different repository". Can we make two workflows one for "self test" and one for "build a different repository"?

Reply:
okay I have a working example I'm ready to push, but one quick question. Right now I am using the format:

```bash
jupyter-repo2docker --debug --user-name username --user-id 1000 --no-run --image-name <container> <repository>
```
 to build, but this of course would probably clone the Github repo defined as <repository> which isn't what we want to do - we want to build where we are during the PR, the stuff in the $PWD! So what I'm hoping is the case is that if the repository isn't defined (e.g., we leave the variable blank) then it will just build what it finds in $PWD? So I would do this:

```bash
jupyter-repo2docker --debug --user-name username --user-id 1000 --no-run --image-name <container>
```

Reply:
okay I'm kind of excited to see if this works! I've just added an example notebook and requirements.txt... See the notes at the top of the file:

```bash
  # If you want to deploy your container to Docker Hub, you must
  # define environment variables in the CircleCI web interface for your project
  #  1. define the CONTAINER_NAME environment variable for the project
  #     Specifically, this is the name of the Docker Hub <ORG>/<NAME> to push
  #     If not defined, will use the repository that is being built from.
  #     to in the case that you have provided Docker credentials. 
  #  2. Define DOCKER_USER and DOCKER_PASS (for push to Docker Hub, if wanted)
  #     If you don't define these in CircleCI settings it won't be pushed.
  #  3. Optionally define the DOCKER_TAG. If not defined, will use
  #     the commit associated with the build.
  #  4. By default, the build assumes wanting to use the local notebook here.
  #     If you want to build from a **different** Github repository, then define
  #     the variable REPO_NAME as the name of the repository that you 
  #     want to build. That can be done in this file under build:environment,
  #     or again online in the CircleCI environment variable settings. 
```
A little verbose, but it gets the idea across. If the user doesn't define anything, it tries building that repository. With no DOCKER_USER or DOCKER_PASS it just builds (no push to Docker Hub). If the user defines a `REPO_NAME` then it builds that one instead.